cpp/src/strings/copying/concatenate.cu

```diff
@@ -41,50 +41,66 @@ std::unique_ptr<column> concatenate( std::vector<strings_column_view> const& str
     CUDF_EXPECTS( strings_count < std::numeric_limits<size_type>::max(), "total number of strings is too large for cudf column" );
     if( strings_count == 0 )
         return make_empty_strings_column(mr,stream);
-    size_t total_bytes = thrust::transform_reduce( strings_columns.begin(), strings_columns.end(),
-        [] (auto scv) { return scv.chars_size(); }, static_cast<size_t>(0), thrust::plus<size_t>());
+
+    // build vector of column_device_views
+    std::vector<std::unique_ptr<column_device_view,std::function<void(column_device_view*)> > > device_cols(strings_columns.size());
+    thrust::host_vector<column_device_view> h_device_views;
+    for( auto &scv : strings_columns )
```

Comment:
Using forwarding (universal) reference `&&` in: 
``` for( auto&& scv : strings_columns )``` 
instead of l-value reference (`&`) is typically more efficient, because it can avoid unnecessary copies.